[Dexter_Wong] iP

Dexter-Wong opened the pull request:
# Duke IP by Dexter Wong 
> Wow, I really wished that I have a simple to use, all in one task manager! - Every Software Engineer

## Fear not!  This project supports:
- Todo list tracking
- Deadline managers
- FInd and search features
- And more!

All you need to do is:
1. Download the zip file
2. Extract and run the file
3. Enjoy!

- [x] Self made
- [x] Fully functional
- [x] Track your food recipies 😋 

**UI coming soon!**

Some basic commands are:

```ruby
todo task 1

deadline task 2 /by 5th Jan

event lecture /from Monday 4pm /to Monday 5pm 
```

You will never miss an `Event` ever again!

Find my GitHub [here](https://github.com/Dexter-Wong)!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Dexter-Wong committed:
Level 0 commit

Dexter-Wong committed:
Level-1

Dexter-Wong committed:
1.5

Dexter-Wong committed:
Level-2, added ArrayList support for adding and listing tasks

Dexter-Wong committed:
Level-3
Added functionality to mark and unmark tasks

Dexter-Wong committed:
Level-4
Added functionality to Todo, Deadline, and Event commands

Dexter-Wong committed:
A-TextUiTesting
Script for auto testing implemented

Dexter-Wong committed:
4.5 laptop to pc

Dexter-Wong committed:
Level-5
Errors handled for Unknown commands, Invalid use of Mark, Todo, Deadline and Event. Exceptions added

Dexter-Wong committed:
Level-6
Added delete functionality and updated exception handling

Dexter-Wong committed:
Add new updates to methods to increase code functionality

Dexter-Wong committed:
Add functionality for saving the task list on every command as well as loading from the local hard disc

The expansion of the chatbot comes with new features on every commit. As such, there will need to be more OOP practices within the chatbot.

Let's tody up the current code by creating new classes and packages for the Fireraya chatbot

Dexter-Wong committed:
Add comment for outdated backup file

Dexter-Wong committed:
Add a text file to docs folder

howenyap commented on backup/main/java/fireraya/command/ToDoCommand.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws FirerayaException {
+        Todo todo = new Todo(description);
+        tasks.add(todo);
+        ui.printAddSuccessMessage(todo, tasks.size());
+        storage.saveToFile(tasks.getTasks());
```
Method and variable names are easily understood, good job!

howenyap commented on backup/main/java/fireraya/main/Fireraya.java:
```diff
@@ -0,0 +1,229 @@
+
+    public static void listTasks() throws IOException {
+        System.out.println("Here is a list of your tasks!");
+    for (int i = 0; i < tasks.size(); i++) {
+        Task current = tasks.get(i);
+        int num = i + 1;
+        System.out.println(num + "." + current.toString());
```
There should be an indentation here

howenyap commented on backup/main/java/fireraya/main/Fireraya.java:
```diff
@@ -0,0 +1,229 @@
+
+
+    public static void main(String[] args) throws FirerayaException {
+
+
+
```
Try to not leave unnecessary space!

howenyap commented on backup/main/java/fireraya/main/Fireraya.java:
```diff
@@ -0,0 +1,229 @@
+
+
+                if (keyword.equals("event")) {
+                    int indexf = -1;
+                    int indext = -1;
```
What do indexf and indext represent here?

howenyap commented on backup/main/java/fireraya/main/Fireraya.java:
```diff
@@ -0,0 +1,229 @@
+                }
+
+                if (keyword.equals("delete")) {
+                    if (all.length > 2) {
+                        throw new InvalidNumOfArgsException();
+                    }
+                    int curr = Integer.parseInt(all[1]) - 1;
+                    if (tasks.size() <= curr || curr < 0) {
+                        throw new FirerayaException("That task does not exist!");
+                    }
+                    System.out.println("Noted. I've removed this task:");
+                    listTask(curr);
+                    tasks.remove(curr);
+                    taskCount();
+                    continue;
```
Try to add space between lines doing a specific task for better readability!
```suggestion
                    if (all.length > 2) {
                        throw new InvalidNumOfArgsException();
                    }
                    
                    int curr = Integer.parseInt(all[1]) - 1;
                    
                    if (tasks.size() <= curr || curr < 0) {
                        throw new FirerayaException("That task does not exist!");
                    }
                    
                    System.out.println("Noted. I've removed this task:");
                    listTask(curr);
                    tasks.remove(curr);
                    taskCount();
                    
                    continue;
```

howenyap reviewed (commented):
Naming conventions are easy to understand, good job!

Dexter-Wong committed:
Add deadline and Date support for the deadline command and update this to reflect in the automatic loading and saving

Dexter-Wong committed:
Merge branch 'branch-level-8'

Dexter-Wong committed:
Remove some import statements.

Trivial commit. This is for tagging purposes of the OOP level

Dexter-Wong committed:
Remove unneeded import statements. Trivial commit to facilitate tagging of Package tag

Dexter-Wong committed:
Merge branch 'add-gradle-support'

Dexter-Wong committed:
Add support for automated builds using Gradle

Dexter-Wong committed:
Add JUnit tests for Todo and Tasklist classes

Dexter-Wong committed:
Trivial commit for purposes of creating executable JAR file

Dexter-Wong committed:
Add find functionality to the current version of app

Dexter-Wong committed:
Merge branch 'branch-level-9'

Dexter-Wong committed:
Test to commit

Dexter-Wong committed:
Commit coding standard

Dexter-Wong committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	docs/List.txt

Dexter-Wong committed:
Add Javadocs for Fireraya class

Dexter-Wong committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	docs/List.txt

baskargopinath commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/Fireraya.java` file, right-click it, and choose `Run Fireraya.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
Would it be possible to summarise this further?

baskargopinath commented on backup/main/java/fireraya/main/Fireraya.java:
```diff
@@ -0,0 +1,229 @@
+
+                if (keyword.equals("unmark")) {
+                    if (all.length > 2) {
+                        throw new InvalidNumOfArgsException();
```
You can consider using switch instead of if else

baskargopinath commented on backup/main/java/fireraya/task/Todo.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    @Override
+    public String saveFormat() {
```
Method names looks good and are easy to understand

baskargopinath commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,28 @@
+[T][ ] a
+Now you have 1 tasks in the list
+Here is a list of your tasks!
+1.[T][ ] a
```
Looks good and functional!

baskargopinath commented on src/main/java/fireraya/task/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+    @Override
+    public String saveFormat() {
+        if (by == null) {
+            DateFormat df = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss");
```
Very apt and understandable method names

baskargopinath reviewed (commented):
LGTM. Just some minor suggestions. Hope its helpful!

Dexter-Wong committed:
Add updated Javadocs for Fireraya Class

Dexter-Wong committed:
Add basic gui to the application

Dexter-Wong committed:
Add functionality for another Assert test

There are several methods updated with the increase of the Fireraya chatbot

To make sure that we have not have had code regression, let's add a few more assert tests

Dexter-Wong committed:
Merge branch 'branch-A-Assertions'

Dexter-Wong committed:
Add additional functionality to the GUI

Before, we had classes which did not have enough OOP principles used. Now we furthur divide the GUI aspects into sub classes

Dexter-Wong committed:
Add functionality for the chatbot to automatically exit the app when exit command is typed

Having to manually close the app is troublesome. Let's add a way for the chatbot to close automatically upon giving a bye command!

Dexter-Wong committed:
Modify the maximum height limit for the mainwindow fmxl class

When using list feature, all your tasks would not be displayed
originally as there were too many tasks to be shown.

We will work around this by removing the prefheight limit.

Dexter-Wong committed:
Add assertion checks

We will be adding an assertion check for the find command.

Dexter-Wong committed:
Add Javadocs comments for many of the methods in the program

Adding methods to the Main, Task, Ui and Exception packages allows
users to more easily understand the content and the code inside.

Dexter-Wong committed:
Add Javadocs comments to the command packages

Having these Javadocs comments completes most of the program's
documentation.

Dexter-Wong committed:
Merge pull request #2 from Dexter-Wong/branch-A-CodeQuality

Branch a code quality

Dexter-Wong committed:
Add the ability to use a DoAfter command

With some tasks having specific requirements of needing to be done
after a certain date, we have implemented a new type of
command which allows one to put up a task to do after a certain date.

Dexter-Wong committed:
Merge branch 'branch-BCD-Extension'

Dexter-Wong committed:
Add the UI photo to the docs list

We will add the UI photo to the docs list for display purposes.
Additionally we will delete unneeded files in the docs folder.

Dexter-Wong committed:
Update the user guide to include a brief overview of commands

We also have updated the build.gradle file to reflect the main
class of the application. Code quality has also been improved.

Dexter-Wong committed:
Add a final release version for the chatbot

We have fixed some classpath issues for the executable jar file
and it should be able to run now.

Dexter-Wong committed:
Remove backups and improve error handling for users

The previous version did not support some errors which were
present in the current version of the application. This new release
will update and fix those issues.

Dexter-Wong committed:
Add a new update to better handle errors in the delete function

The previous delete feature did not account for non inputs. This change
lets users know what to do in the event of an invalid input.

Dexter-Wong committed:
Tidy up code and remove one layer of abstraction in the parser class

The parser class was originally composed of a try block, which has been
removed to reduce one layer of abstraction since the main class
already has a try catch block.
